### PR TITLE
Devkit architecture

### PR DESCRIPTION
**Pull request by milstan**
For working on the Devkit Architecture document

### CODE REVIEW

**Comment by milstan** [comment]
I've made a sub-branch with few minor changes I proposed under [this pull request](https://github.com/blindnet-io/product-management/pull/722).

Other discussion points:
- The following components should be better defined (what they are + see comment about "rights" under "Privacy rights engine")
    > Data consumption components
    > Data rights capture components
    > Data rights settings component
    > Data rights management components

- Data capture engine

    - Should the [*Capture Component* from HLA](https://github.com/blindnet-io/product-management/tree/devkit-architecture/refs/high-level-architecture#capture-component) be renamed to match this name?
    
- Data encryption engine

    - Maybe just call it Encryption Engine (as the subject of encryption is obviousely the data)
    
    - Should [*Encryption and Access Management Engine* from HLA](https://github.com/blindnet-io/product-management/tree/devkit-architecture/refs/high-level-architecture#encryption-and-access-management-engine) be renamed to match this terminology? If those two are the same thing, they should be called the same across all documents. If they are not the same, then the difference should be clearly adressed here.

- Privacy rights engine

    - I am trying to move away from the word "rights" in naming of components and functions. This is because, fundamentally privacy IS NOT about (human) rights, but [it is about connectedness. ](https://github.com/blindnet-io/product-management/blob/devkit-architecture/refs/notion-of-privacy/notion-of-privacy.md). What we do here is allow the Users and System regulate their connectedness, not respect rights. A User may make a request, and System may grant it even when the user does not have the legal right - but just a possibility. I don't want developers implementing this to focus on "rights" as this is misleading and might ultimately lead to a disfunctional implementation of privacy in Software Systems. I think, "rights" ARE the very reason why Privacy is so poorly implemented on the internet. So I would very much recommend to avoid using the word "rights" whenever possible. Maybe call it Privacy Engine?
    
    - What is the link between this and [*Privacy Computation Engine* from HLA](https://github.com/blindnet-io/product-management/tree/devkit-architecture/refs/high-level-architecture#privacy-computation-engine)? Either change the name in HLA and elsewhere in the documents, or explain the difference.

- In general:
    - It should be very clear what part of this architecture corresponds to what part of the HLA. 
        - Make changes to HLA if needed, to acheive coherence. 
        - There should not be components of the HLA that are not mapped to something in this document. (e.g. Privacy Compiler is not mentioned)
    - Please note that Data Rights Request is now called Privacy Request.

**Comment by filipblnt** [comment]
> Should the [Capture Component from HLA](https://github.com/blindnet-io/product-management/tree/devkit-architecture/refs/high-level-architecture#capture-component) be renamed to match this name?

No, same for others.

File: specifications/architecture/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+*Q3 2022 Scope*:
+- Set relevant configurations (e.g., Retention Policies, General Information for TRANSPARENCY requests, Transfers information, desired level of automation of Privacy Request Processing )
+- Manage Privacy Requests (View Privacy Requests, Act upon Privacy Request Response recommendations - grant/deny/transfer requests)
+
+*Autumn 2022 Scope*:
+- View and manage Data Captures
+
+**Web components** are front-end, look and feel agnostic components which allow integrations of different blindnet devkit functions into external systems and web sites. Currently, these include:
+- Login component (System must be able to work with existing login engines of client Systems)
```
Comment (milstan) [inline specifications/architecture/README.md:36]: Can we find a way to not make the login component ourselves at all, but fully rely on client's login solution or an open-source component?

**Comment by milstan** [comment]
@TheKinrar please review and make sure this is in synch with recent discussions and with your PR #748 

@noelmace please act as Assignee here, as Filip has too much work currently.

**Comment by noelmace** [comment]
Adding @m4rk055 as a reviewer as @TheKinrar this out of offsite/
I'll merge this on Jully 7th except @m4rk055 has any formal objection.
Please add your complete review ASAP.

File: specifications/architecture/README.md (hunk 1 above)
Comment (m4rk055) [inline specifications/architecture/README.md:36]:
```suggestion
```

File: specifications/architecture/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+- Allows recovery after access is lost
+
+The [HLA][HLA]'s **Privacy Computation Engine**, including **Privacy Compiler** and **Customization API**, consists of _Privacy SDK_, _Privacy API_, and _Privacy DB_, and it:
+- Captures Privacy Requests from Data Subjects
+- Captures privacy-related Settings (including mapping between PRIV terms and database schema of the client System)
+- Interprets Data Capture metadata (based on Settings)
+- Calculates (explainable) response to Privacy Requests
+- Keeps traces of Privacy Requests decisions and actions
+- Provides proofs of Privacy Requests decisions and actions
+- Registers operations and transfers of Data Captures across Systems
```
[the remark this answers is not in the document]
Comment (m4rk055) [reply specifications/architecture/README.md:75]: Transfers should be done by the capture or storage component

[the remark this answers is not in the document]
Comment (milstan) [reply specifications/architecture/README.md:73]: @m4rk055 does @TheKinrar know that transfers are in Capture/Storage component?

[the remark this answers is not in the document]
Comment (m4rk055) [reply specifications/architecture/README.md:73]: I misread it as it the PCE component **transfers** the data, not **registers the transfers**.
We haven't specified the transfers yet but @TheKinrar should be aware of it.

**Review by m4rk055 — APPROVED** [review]

**Pull request merged**